### PR TITLE
Warning fixes

### PR DESCRIPTION
**Pull request by seanm**

### CODE REVIEW

File: src/mat.c [hunk 1]
```diff
@@ -69,26 +69,29 @@
     (sizeof(MAT_MKTEMP_DIR) + sizeof(MAT_MKTEMP_TPL) + sizeof(MAT_MKTEMP_FILE) - 2)
 
 static char *
-Mat_mktemp(char *path_buf, char *dir_buf)
+Mat_mktemp(char *path_buf, size_t path_buf_size, char *dir_buf, size_t dir_buf_size)
```
Comment (tbeu) [inline src/mat.c:72]: It might be good enought to assume `MAT_MKTEMP_BUF_SIZE` for both buffers to avoid passing buffer sizes.

File: src/mat.c [hunk 2]
```diff
@@ -1290,7 +1294,7 @@ Mat_VarDelete(mat_t *mat, const char *name)
     if ( NULL == mat || NULL == name )
         return err;
 
-    if ( NULL != Mat_mktemp(path_buf, dir_buf) ) {
+    if ( NULL != Mat_mktemp(path_buf, MAT_MKTEMP_BUF_SIZE, dir_buf, MAT_MKTEMP_BUF_SIZE) ) {
```
Comment (tbeu) [inline src/mat.c:1297]: See above.

File: src/mat5.c [hunk 3]
```diff
@@ -2103,6 +2103,7 @@ WriteType(mat_t *mat, matvar_t *matvar)
                     for ( j = nBytes % 8; j < 8; j++ )
                         fwrite(&pad1, 1, 1, (FILE *)mat->fp);
             }
+            break;
```
Comment (tbeu) [inline src/mat5.c:2106]: Luckily there was no damage in not having the `break` here.

**Review by tbeu — COMMENTED** [review]
Thanks for the PR. Very much appreciated.

File: src/mat.c (hunk 1 above)
Comment (seanm) [reply src/mat.c:72]: If you prefer, I can make that change.  But it's not like the extra param is going to be slower than the file system interaction...

File: src/mat5.c (hunk 3 above)
Comment (seanm) [reply src/mat5.c:2106]: Yeah!

File: src/mat.c (hunk 1 above)
Comment (tbeu) [reply src/mat.c:72]: Yes, please go ahead. There is no strong reason to change the declaration of Mat_mktemp.

Be aware that I pushed to your branch.

Comment (seanm) [reply src/mat.c:72]: >Be aware that I pushed to your branch.

My git-foo is weak.  Presumably this means I should get your changes to my branch, right?  Will have to google that...

Comment (tbeu) [reply src/mat.c:72]: Just git pull on your branch.

Comment (seanm) [reply src/mat.c]: OK done, hopefully I haven't clobbered whatever you changed...

Comment (seanm) [reply src/mat.c]: @tbeu did you have a chance to look this over again?

**Comment by tbeu** [comment]
Unfortunately, the CI setup is rather fragile. At least, it is running for Travic CI again, bit still need to fix it for Appveyor CI. Once I know how to address https://ci.appveyor.com/project/tbeu/matio/builds/39979026/job/78insi47y9l4hpu7#L166 I will rebase your branch again and merge afterwards. Thanks for your contribution and patience.

**Comment by seanm** [comment]
Thanks a lot!

**Pull request merged**